src/passes/pass.cpp

```diff
@@ -594,18 +594,24 @@ void PassRunner::addDefaultGlobalOptimizationPrePasses() {
     addIfNoDWARFIssues("once-reduction");
   }
   if (wasm->features.hasGC() && options.optimizeLevel >= 2) {
-    addIfNoDWARFIssues("type-refining");
-    addIfNoDWARFIssues("signature-pruning");
-    addIfNoDWARFIssues("signature-refining");
+    if (options.closedWorld) {
```

Comment:
What would you think about moving this logic to check `closedWorld` into each pass, just like we used to bail out of passes early under the equirecursive type system? That will be closer to how we want this to work eventually, where passes will still run under an open world but will be less aggressive.

Reply:
That's possible, but it would be harder to test. The current test checks if the passes are run at all. If instead the passes ran but exited early then we'd need specific tests for each pass to see if they have an effect or not. That's not unreasonable, but I'm not sure it's worth the extra work at this time?

Reply:
Ok, doing this work later seems fine as well.